core/src/main/kotlin/org/evomaster/core/problem/rest/service/AbstractRestSampler.kt

```diff
@@ -133,6 +141,36 @@ abstract class AbstractRestSampler : HttpWsSampler<RestIndividual>() {
         log.debug("Done initializing {}", AbstractRestSampler::class.simpleName)
     }
 
+    /*
+    This function retrieves the swagger. It is used for both black-box and white-box.
+     */
+    private fun retrieveSwagger(openApiURL : String) {
+
+        // first try to retrieve the OpenAPI without authentication
+        try {
+            swagger = OpenApiAccess.getOpenAPIFromURL(openApiURL, HttpWsNoAuth())
+        }
+        catch (sutException : AuthenticationRequiredException) {
+            log.warn(sutException.message)
+
+            // First check if we have authentication information available inside infoDto.infoForAuthentication
+            if (authentications.isNotEmpty()) {
+
+                //get the first authentication info
+                val currentAuthInfo = authentications.getFirstAuthentication()
+
+                // try to retrieve the swagger with authentication info
+                swagger = OpenApiAccess.getOpenAPIFromURL(openApiURL, currentAuthInfo)
+            }
```

Comment:
add `else` branch which throws a new `AuthenticationRequiredException`, with error message stating that accessing schema requires authentication, but there is no auth info provided that can be used

Reply:
Hi Andrea,

Thanks for your comment. I have fixed this and I have also added another e2e test file testing this case.